src/game/features/self/EagleEye.cpp

```diff
@@ -0,0 +1,33 @@
+#include "core/commands/LoopedCommand.hpp"
+#include "game/features/Features.hpp"
+#include "game/rdr/Enums.hpp"
+#include "game/rdr/Natives.hpp"
+#include "core/frontend/Notifications.hpp"
+
+namespace YimMenu::Features
+{
+	class EagleEye : public LoopedCommand
+	{
+		using LoopedCommand::LoopedCommand;
+
+		virtual void OnTick() override
+		{
+			if(!PLAYER::_IS_SECONDARY_SPECIAL_ABILITY_ACTIVE(PLAYER::PLAYER_ID())) // Checks if Eagle Eye is active before toggling it on
```

Comment:
I think this could be modified to detect when they are in eagle eye and just give them unlimited then

Reply:
> I think this could be modified to detect when they are in eagle eye and just give them unlimited then

removing the if statement and its innards does that on its own, but it wont auto activate for you.  But yeah it works that way by removing this

```if(!PLAYER::_IS_SECONDARY_SPECIAL_ABILITY_ACTIVE(PLAYER::PLAYER_ID())) // Checks if Eagle Eye is active before toggling it on
            {
                PLAYER::_SECONDARY_SPECIAL_ABILITY_SET_ACTIVE(PLAYER::PLAYER_ID()); // Toggles Eagle Eye on
                Notifications::Show("Eagle Eye", "Infinite Eagle eye has been enabled.", NotificationType::Success);
                Notifications::Show("WARNING", "When this is active, it will always reactivate eagle eye for you, you still cannot run using it.", NotificationType::Warning);
            }```

Reply:
Only using toggle for `PLAYER::_MODIFY_INFINITE_TRAIL_VISION(PLAYER::PLAYER_ID(), 1); // Sets Eagle Eye to Infinite` will toggle infinite eagle eye on/off, youd still have to activate it on your own.